lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -885,34 +885,58 @@ static void optimizePool(Function *F) {
   } // For all nodes in the graph.
 }
 
-/// \returns The uniquely used Constant from node or nullptr
-/// if node has more than one user or is not a Constant.
-static Constant *getUniquelyUsedConstant(Node &node) {
-  // If that node has more than one use, it may not
-  // be okay to modify the underlying Constant.
-  if (!node.hasOneUse()) {
+/// \returns a uniquely used Constant with the same contents as \p node. If \p
+/// node is not a Constant then \returns a nullptr. If \node is a Constant which
+/// has a single use, \p node is returned. If \node is a Constant which has
+/// multiple uses, then \returns a new duplicate Constant that has the same
+/// contents as \p node.
+static Constant *getUniquelyUsedConstant(Function *F, Node &node) {
```

Comment:
Nit: given we just use F->getParent(), maybe we should pass the module instead.

Reply:
Done